### ISSUE TITLE
Best practice for extending Admin::User model

### ISSUE THREAD

**Issue by mattr** [katalyst/koi#574]
The `Admin::User` model from Koi is not directly present in the generated application. This leads to potential issues if we need to extend the model for app-specific behaviour.

I see three main avenues for extending the model:
1. Copy the model from Koi into the application wholesale;
2. Extend the model in the application; or
3. Extend the model via initializers.

Each option has benefits and drawbacks and may be suitable in different situations. For example, 3 might be a preferred approach if we are just adding a concern to the model. 1 is the simplest solution but 2 reduces the additional boilerplate.

We want to document our preferred approach(es) and in which situations they would be most applicable.

**Comment by sfnelson** [comment]
I don't think this is really an intended use case in Koi. As we've discussed Koi is not intended to be a seed for a white-label product, it's a light-weight admin to add to an existing website. Supporting extension points in the admin user model in particular makes it much harder to add new features to Koi itself.

Not a project goal at present, happy to revisit if there's a compelling use case in the future.
